server/datastore/mysql/policies.go

```diff
@@ -54,6 +54,26 @@ func (ds *Datastore) Policy(ctx context.Context, id uint) (*fleet.Policy, error)
 	return policyDB(ctx, ds.reader(ctx), id, nil)
 }
 
+func (ds *Datastore) PolicyByName(ctx context.Context, name string) (*fleet.Policy, error) {
+	var policy fleet.Policy
+	err := sqlx.GetContext(ctx, ds.reader(ctx), &policy,
+		fmt.Sprint(`SELECT `+policyCols+`,
+		COALESCE(u.name, '<deleted>') AS author_name,
+		COALESCE(u.email, '') AS author_email,
+		(select count(*) from policy_membership where policy_id=p.id and passes=true) as passing_host_count,
+		(select count(*) from policy_membership where policy_id=p.id and passes=false) as failing_host_count
+		FROM policies p
+		LEFT JOIN users u ON p.author_id = u.id
+		WHERE p.name=?`), name)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, err
```

Comment:
```go
if err == sql.ErrNoRows {
  return nil, ctxerr.Wrap(ctx, notFound("Policy").WithName(name))
}
```
Similar to:
https://github.com/fleetdm/fleet/blob/700fdd39999ca52beee8f16fefd82ad9d771d6b8/server/datastore/mysql/policies.go#L78

Reply:
And on the caller side you'd use `fleet.IsNotFound`

Reply:
TY, updated.